raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -746,9 +764,9 @@ public void testCannotResignWithLargerEpochThanCurrentEpoch(boolean withKip853Rp
     @ParameterizedTest
     @ValueSource(booleans = { true, false })
     public void testCannotResignIfNotLeader(boolean withKip853Rpc) throws Exception {
-        int localId = 0;
-        int otherNodeId = 1;
-        int leaderEpoch = 2;
+        int localId = randomReplicaId();
+        int otherNodeId = localId + 1;
+        int leaderEpoch = localId + 2;
```

Comment:
The leader epoch should stay as 2. No need to randomize the leader epoch for now. E.g. `int leaderEpoch = 2;`.

Reply:
yep, nice catch! updated in latest commit